OrbitGl/CallStackDataView.cpp

```diff
@@ -13,7 +13,7 @@
 
 using orbit_client_protos::FunctionInfo;
 
-CallStackDataView::CallStackDataView() : DataView(DataViewType::kCallstack), callstack_(nullptr) {}
+CallStackDataView::CallStackDataView(DataViewType type) : DataView(type) {}
```

Comment:
I'm not too happy with this change, even though I get that it's consistent. I'm fine with leaving it like that since it's more of a conceptual problem with the DataViewFactory, but let me put my thoughts down here nevertheless.

If I have a factory, I expect it to create objects of different types and behaviors (if only slighty) when I pass in different enum value. So when I look at this, I expect a different type of object being constructed for kCallstack and kSelectionCallstack, or at least an object that is initialized in a different way, resulting in - even if slightly - different behavior. This is not the case the Callstack - here it is basically used to create two instances of the same type. For the CallStackDataView, it is completely irrelevant that it is of type kSelectionCallStack. 

As I said, bit of a higher-level conceptual problem - to me, it does not make sense to have a factory with a "GetOrCreate" method. A factory's functionality shouldn't be to manage singleton-like objects. 

In the end, this factory interface is implemented exactly once, in OrbitApp. It could very well be replaced with `OrbitApp::GetOrCreateProcessesView()`, `OrbitApp::GetOrCreateModulesView()`. Or with a `OrbitDataViewCollection` class that does this as a member of OrbitApp, if we want to keep the modularization.

Edit: I've just seen that not all of the DataViewTypes are even supported by the factory... all the more reason to maybe remove it alltogether?